proposals/nnnn-tuple-of-value-pack-expansion.md

```diff
@@ -0,0 +1,88 @@
+# Tuple of value pack expansion
+
+* Proposal: [SE-NNNN](nnnn-tuple-of-value-pack-expansion.md)
+* Authors: [Sophia Poirier](https://github.com/sophiapoirier), [Holly Borla](https://github.com/hborla)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+* Upcoming Feature Flag: `VariadicGenerics`
+* Previous Proposal: [SE-0393](0393-parameter-packs.md)
+
+## Introduction
+
+Building upon the **Value and Type Parameter Packs** proposal [SE-0393](https://forums.swift.org/t/se-0393-value-and-type-parameter-packs/63859), this proposal enables referencing a tuple value that contains a value pack inside a pack repetition pattern.
+
+## Motivation
+
+When a tuple value contains a value pack, there is no way to reference those pack elements or pass them as a value pack function argument.
+
+Additionally, type parameter packs are only permitted within a function parameter list, tuple element, or generic argument list. This precludes declaring a type parameter pack as a function return type, type alias, or as a local variable type to permit storage. The available solution to these restrictions is to contain the value pack in a tuple, which makes it important to provide full functional parity between value packs and tuple values containing them. This proposal fills that functionality gap by providing a method to reference individual value pack elements contained within a tuple.
+
+## Proposed solution
+
+This proposal extends the functionality of pack repetition patterns to values of _abstract tuple type_, which enables an implicit conversion of an _abstract tuple value_ to its contained value pack. An _abstract tuple type_ is a tuple that has an unknown length and elements of unknown types. Its elements are that of a single type parameter pack and no additional elements, and no label. In other words, the elements of the type parameter pack are the elements of the tuple. An _abstract tuple value_ is a value of _abstract tuple type_. This proposal provides methods to individually access the dynamic pack elements of an abstract tuple value inside of a repetition pattern.
```

Comment:
Can you expand on this slightly--perhaps an example or two to illustrate use of the proposed feature? Ideally, these examples would also illustrate what isn't permitted (in terms of labels, etc.).

I think reviewers are going to be interested to know what this proposal will allow them or us to do. For example, will it make it possible to implement a variadic `ZipSequence` or variadic tuple `==`?

Reply:
Things like `ZipSequence` and tuple equality are already possible with the main feature proposal, however you become stuck with being able to do anything with a pack's elements after a value pack is forced into a tuple in order to be returned from a function or stored to a variable. So this is less about enabling the implementation of some function, but more what you can do next when you call any function that returns value packs. I will clarify this and add examples (including invalid ones).